File: Chapter05/Java/StorageExample1/app/src/main/java/net/zenconsult/mobile/RetrieveData.java
package net.zenconsult.mobile;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Hashtable;

public class RetrieveData {

    public static Hashtable<String, Object> get(Context ctx) {
        String hostname = "hostname";
        String port = "port";
        String ssl = "ssl";

        Hashtable<String, Object> data = new Hashtable<>();
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
        data.put(hostname, prefs.getString(hostname, null));
        data.put(port, prefs.getInt(port, 0));
        data.put(ssl, prefs.getBoolean(ssl, true));
        return data;
    }

}

File: Chapter05/Kotlin/StorageExample1/app/src/main/java/net/zenconsult/mobile/RetrieveData.kt
package net.zenconsult.mobile

import android.content.Context
import androidx.preference.PreferenceManager
import java.util.*

class RetrieveData {

    companion object {
        fun get(context: Context): Hashtable<String, Any?> {
            val hostname = "hostname"
            val port = "port"
            val ssl = "ssl"

            val data = Hashtable<String, Any?>()
            val prefs = PreferenceManager.getDefaultSharedPreferences(context)
            data[hostname] = prefs.getString(hostname, null)
            data[port] = prefs.getInt(port, 0)
            data[ssl] = prefs.getBoolean(ssl, true)
            return data
        }
    }

}